contracts/managers/SocialTradingManager.sol

```diff
@@ -0,0 +1,296 @@
+/*
+    Copyright 2019 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { ICore } from "set-protocol-contracts/contracts/core/interfaces/ICore.sol";
+import { ILiquidator } from "set-protocol-contracts/contracts/core/interfaces/ILiquidator.sol";
+import { IRebalancingSetTokenV2 } from "set-protocol-contracts/contracts/core/interfaces/IRebalancingSetTokenV2.sol";
+import { ISetToken } from "set-protocol-contracts/contracts/core/interfaces/ISetToken.sol";
+import { RebalancingLibrary } from "set-protocol-contracts/contracts/core/lib/RebalancingLibrary.sol";
+
+import { ISocialAllocator } from "./allocators/ISocialAllocator.sol";
+
+
+/**
+ * @title SocialTradingManager
+ * @author Set Protocol
+ *
+ * Singleton manager contract through which all social trading sets are managed. Traders can choose a percentage
+ * between 0 and 100 for each rebalance. The trading pair used for each trading pool is defined by the allocator
+ * passed in on pool creation. Only compatible with RebalancingSetTokenV2 constracts. All permissioned functions
+ * on the RebalancingSetTokenV2 must be called through the administrative functions exposed on this contract.
+ */
+contract SocialTradingManager {
+    using SafeMath for uint256;
+
+    /* ============ Structs ============ */
+    struct PoolInfo {
+        address trader;
+        ISocialAllocator allocator;
+        uint256 currentAllocation;
+    }
+
+    /* ============ Events ============ */
+
+    event TradingPoolCreated(
+        address indexed _trader,
+        ISocialAllocator indexed _allocator,
+        address _tradingPool,
```

Comment:
This could just have a type of `IRebalancingSetTokenV2` - since there wouldn't need to be conversions later

Reply:
Well this is emitted when we call create on `Core` which returns an address. And since we can't save in the mapping as IRebalancingSetTokenV2 defining the output here as such would actually be converting something that doesn't really need to be converted.